File: src/CountPrimes.java
public class CountPrimes {

    public static int countPrimes(int number) {
        boolean[] sieve = getSieve(number);
        return numberOfPrimes(sieve);
    }

    private static int numberOfPrimes(boolean[] array) {
        int count = 0;
        for (boolean value : array) {
            if (!value) count++;
        }
        return count;
    }

    private static boolean[] getSieve(int number) {
        if (number < 2) {
            return new boolean[0];
        }

        boolean[] sieve = new boolean[number];
        sieve[0] = sieve[1] = true;
        for (long index = 2 ; index < sieve.length ; index++){
            if (!sieve[(int) index]) {
                for (long j = index * index ; j < sieve.length ; j += index) {
                    sieve[(int) j] = true;
                }
            }
        }

        return sieve;
    }
}

File: python/count_primes.py
from typing import List


class Solution:
    def _get_prime_sieve(self, size: int) -> List[int]:
        sieve = [1] * (max(size, 2))
        sieve[0], sieve[1] = 0, 0
        for number in range(2, len(sieve)):
            if sieve[number]:
                for dividend in range(number ** 2, len(sieve), number):
                    sieve[dividend] = 0
        return sieve

    def countPrimes(self, n: int) -> int:
        return sum(self._get_prime_sieve(n))
